### PR TITLE
[Training] [1/n] Add latent datasets

### PR DESCRIPTION
**Pull request by SolitaryThinker**
- [x] Clean up code

cc @JerryZhou54

### CODE REVIEW

**Comment by SolitaryThinker** [comment]
rank 0:
	1. first reads in # rows pers per parquet file, # of parquet files
	2. Generates list of indexes for each worker (dp, sp), global shuffling
	3. write this “plan” to disk

rank N:
Read plan from disk

Index into its own list of batches (file #, row #)
Read row
latent: [bs, c, t, h, w]
n = sp_size
[bs, c, t, h, w] -> [bs, c, (n s), h, w]  -> [bs, c, t // sp_size, h, w]
latent = [:, :, rank, :, :, :]

**Comment by SolitaryThinker** [comment]
cc @zhisbug

File: fastvideo/v1/dataset/latent_datasets.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    for latent, prompt_embed, latent_attn_mask, prompt_attention_mask in dataloader:
+        print(
+            latent.shape,
+            prompt_embed.shape,
+            latent_attn_mask.shape,
+            prompt_attention_mask.shape,
+        )
+        import pdb
+
+        pdb.set_trace()
```
Comment (Edenzzzz) [inline fastvideo/v1/dataset/latent_datasets.py:129]: It's better to remove or use a separate file for testing.

File: fastvideo/v1/dataset/parquet_datasets.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+
+    # Create DataLoader with proper settings
+    dataloader = StatefulDataLoader(
+        dataset,
+        batch_size=args.batch_size,
+        num_workers=1,  # Reduce number of workers to avoid memory issues
+        prefetch_factor=2,
+        shuffle=False,
+        pin_memory=True,
+        drop_last=True)
```
Comment (Edenzzzz) [inline fastvideo/v1/dataset/parquet_datasets.py:270]: The best practice may be  `pin_memory_device=f"cuda:{torch.cuda.current_device()}`, as latents don't take much memory. The `tensor.to` is blocking on its own.
